### PR TITLE
fix property enforcement

### PR DESCRIPTION
**Pull request by 9rnsr**

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -9223,3 +9223,5 @@ if (Rs.length > 1 &&
     m.popFront;
     assert(m.empty);
 }
+
+package T* addressOf(T)(ref T val) { return &val; }
```
Comment (WalterBright) [inline std/range/package.d]: I don't understand the purpose of this.

Comment (WalterBright) [inline std/range/package.d]: Also, what does adding a new function have to do with property enforcement? Should it be a separate PR? The rest of this PR looks good to me.

Comment (9rnsr) [inline std/range/package.d]: It's necessary to take an address of returned reference from getter property.

1. After the property enforcement finished, a getter property cannot be called with parenthesis: `prop()` -> `prop`
2. But applying address operator `&` on property function name returns the address of "property function": `&prop` -> function pointer or delegate

I had proposed adding a special syntax difference between `&prop` and `&(prop)`, but it was rejected.
So, I think we should use the helper function like `addressOf(prop)` or `prop.addressOf` for the purpose.

Comment (WalterBright) [inline std/range/package.d]: The function needs to be documented.

**Comment by quickfur** [comment]
Similar functions already exist in other parts of Phobos, namely `std/parallelism.d` and `std/algorithm/internal.d`. We should factor all of these out into a common place, instead of copy-n-pasting them into every module/package that needs to take an address of a property.

File: std/range/package.d (hunk 1 above)
Comment (quickfur) [inline std/range/package.d]: Similar functions already exist in other parts of Phobos, namely `std/parallelism.d` and `std/algorithm/internal.d`. We should factor all of these out into a common place, instead of copy-n-pasting them into every module/package that needs to take an address of a property.

**Comment by quickfur** [comment]
ping @9rnsr @WalterBright 
This has been sitting here idle for a long time. Should we go ahead with this, or should we just close it?

**Comment by DmitryOlshansky** [comment]
Ping @9rnsr @WalterBright again

**Comment by 9rnsr** [comment]
Rebased on master.

**Comment by DmitryOlshansky** [comment]
-property was destroyed by https://github.com/D-Programming-Language/dmd/pull/4955 so closing this
